internal/set.go

```diff
@@ -30,3 +36,176 @@ func (s IntSet) Remove(e int) {
 func (s IntSet) Count() int {
 	return len(s)
 }
+
+// Same determines whether two sets are backed by the same store. In the
+// current implementation using hash maps it makes use of the fact that
+// hash maps (at least in the gc implementation) are passed as a pointer
+// to a runtime Hmap struct.
+//
+// A map is not seen by the runtime as a pointer though, so we cannot
+// directly compare the sets converted to unsafe.Pointer and need to take
+// the sets' addressed and dereference them as pointers to some comparable
+// type.
+func Same(s1, s2 Set) bool {
+	return *(*uintptr)(unsafe.Pointer(&s1)) == *(*uintptr)(unsafe.Pointer(&s2))
```

Comment:
Can we do that later? I'd like to see the performance impact and I'd like to get this sorted out first and have it stabilised. Would you file an issue to revisit this?